app/javascript/components/users/SignInPage.spec.tsx

```diff
@@ -12,62 +15,154 @@ jest.mock('../../legacy_react/src/lib/api/sign_in');
+			mockedWebUserSignIn.postSignIn.mockResolvedValue({id: 1});
+
+			// act puts all of the related React updates for the click event into a
+			// single update. Since fireEvent.click calls some promises, we need to make
+			// the callback a Promise and await on act. If we didn't, our test wouldn't
+			// wait for all the possible React changes to happen at once.
+			await act(async () => {
+				fireEvent.click(button);
+			});
+
+			// was document.location.assign called with redirectUrl?
+			expect(locationAssignSpy).toHaveBeenCalledWith('redirectUrl');
+
+			const error = result.getByTestId('signInPageError');
+			expect(error).toBeEmptyDOMElement();
 		});
 
-		const error = result.getByTestId('signInPageError');
-		expect(error).toBeEmptyDOMElement();
 	});
 
 	it('signIn failed', async () => {
 		expect.hasAssertions();
-		const result = render(<Wrapper><SignInPage/></Wrapper>);
-
-		const button = result.getByTestId('signInButton');
-		// everytime you try to call the User SignIn API in this test, return a
-		// promise which resolves to {id: 1}
-		mockedWebUserSignIn.postSignIn.mockResolvedValue({id: 1});
-
-		// act puts all of the related React updates for the click event into a
-		// single update. Since fireEvent.click calls some promises, we need to make
-		// the callback a Promise and await on act. If we didn't, our test wouldn't
-		// wait for all the possible React changes to happen at once.
-		await act(async () => {
-			fireEvent.click(button);
+
+		locationAssign(async (locationAssignSpy:jest.SpyInstance<void, [url: string]>) => {
+
+			const result = render(<Wrapper><SignInPage redirectUrl={'redirectUrl'}/></Wrapper>);
+
+			const button = result.getByTestId('signInButton');
+			// everytime you try to call the User SignIn API in this test, return a
+			// promise which resolves to {id: 1}
+			mockedWebUserSignIn.postSignIn.mockResolvedValue({id: 1});
+
+			// act puts all of the related React updates for the click event into a
+			// single update. Since fireEvent.click calls some promises, we need to make
+			// the callback a Promise and await on act. If we didn't, our test wouldn't
+			// wait for all the possible React changes to happen at once.
+			await act(async () => {
+				fireEvent.click(button);
+			});
+
+			// was document.location.assign not called?
+			expect(locationAssignSpy).not.toHaveBeenCalledWith('redirectUrl');
+
+			const error = result.getByTestId('signInPageError');
+
+			// Sometimes because of React's rendering cycle, the changes haven't been
+			// made to the HTML by the time we need to expect. waitFor tries some sort
+			// function multiple times until it passes (or 5 seconds have passed)
+			waitFor(() => expect(error).toHaveTextContent("Ermahgerd! We had an error!"));
 		});
+	});
+
+	it('renders signInComponent Correctly', () => {
+		const { getByTestId } = render(<Wrapper><SignInPage redirectUrl={'redirectUrl'}/></Wrapper>);
+		expect(getByTestId("SignInComponent")).toBeTruthy;
```

Comment:
You should run `yarn eslint --fix` before committing and pushing. It will show you errors and give you suggestions on how to fix the errors. As an example, you're not supposed to use `.toBeTruthy` in the code.

It would also tell you that the first line of every test is supposed to be the `expect.assertions(numberOfAssertionsInTest)` or `expect.hasAssertions()`.

Additionally, it mentions that every test must have at least one expectation. This one does but some other ones don't.

Reply:
Hi Eric, I removed the `.toBeTruthy` in the code and added the assertion statements. We are still working on these tests.